File: src/app/models/pet.ts
export interface Pet extends BaseModel {
    name?: string;
    size?: Size;
    birthday?: string;
    animalType?: AnimalType;
    breeds?: string[];
}

export interface Breed extends BaseModel {
    name?: string;
}

export interface BaseModel {
    id?: number;
}

export enum Size {
    Unspecified = 0,
    Small = 1,
    Medium = 2,
    Large = 3
}

export enum AnimalType {
    Unspecified = 0,
    Dog = 1,
    Cat = 2
}

File: PetStoreServer/src/PetStoreServer/Models/Pet.cs
using PetStoreServer.ViewModels;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PetStoreServer.Models
{
    public class Pet : BaseModel
    {
        public string Name { get; set; }
        public DateTime? Birthday { get; set; }
        public Breed Breed { get; set; }
        public string ImageUrl { get; set; }
    }

    public enum Size {
        Unspecified = 0,
        Small = 1,
        Medium = 2,
        Large = 3
    }


}
